lib/vmtranslator/translator.go

```diff
@@ -0,0 +1,141 @@
+package vmtranslator
+
+import (
+	"log"
+	"strconv"
+	"strings"
+)
+
+type AC_TYPE int
+
+const (
+	AC_OP AC_TYPE = iota
+	AC_RELOP
+	AC_VAR
+	AC_JUMP
+)
+
+type AsmCommand_t struct{}
+
+type AsmCommand interface{}
+
+func TranslateCommand(cmd *Command) (string, error) {
+	log.Fatal("unimplemented")
+	switch cmdType := cmd.cmdType; cmdType {
+	case C_PUSH, C_POP:
+		return pushPopToHack(cmd), nil
+	case C_ARITHMETIC:
+		return arithmeticToHack(cmd), nil
+	default:
+		//ERROR
+		return "", err
+	}
+}
+
+func pushPopToHack(command *Command) (hack string) {
```

Comment:
Can this be made 2 separate functions for push and pop?

Reply:
looks like it. might be cleaner to keep them together cuz they're pretty related to eachother